### PR TITLE
Arrays and loops

### PR DESCRIPTION
**Pull request by busin11**
1- Created arrays_and_loops.go file
2- Edited README

### CODE REVIEW

File: arrays_and_loops/README.md [hunk 1]
```diff
@@ -21,4 +21,27 @@ Element 2: 2, Square: 4, Even
...
 Element 5: 5, Square: 25, Odd
-```
\ No newline at end of file
+```
+
+Array Squaring and Even/Odd Identifier
+
+This program demonstrates the use of arrays, loops, and basic mathematical operations in Go. It initializes an array of integers, computes the square of each element, and 
+checks whether the number is odd or even. Additionally, the program prints the square root of each element for reference.
+How It Works
```
Comment (tul1) [inline arrays_and_loops/README.md:30]: can you add how to run it with `test.txt` file you've added?

File: arrays_and_loops/arrays_and_loop.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+func main() {
+	var arr [5]int
+	for i:=0;i<len(arr);i++ {
```
Comment (tul1) [inline arrays_and_loops/arrays_and_loop.go:10]: This isn't the best way to use `for` statement in golang. probably the most golanguintisc way is this one:
```suggestion
	for i := range arr {
```

Can you update the code to use the suggested form?

File: arrays_and_loops/test.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+Element: " Square: %#v %#v" 1%!(EXTRA int=0, float64=0, string=even)Element: " Square: %#v %#v" 2%!(EXTRA int=1, float64=1, string=odd)Element: " Square: %#v %#v" 3%!(EXTRA int=4, float64=2, string=even)Element: " Square: %#v %#v" 4%!(EXTRA int=9, float64=3, string=odd)Element: " Square: %#v %#v" 5%!(EXTRA int=16, float64=4, string=even)Element: 1 0 Square: 0evenElement: 2 1 Square: 1oddElement: 3 4 Square: 2evenElement: 4 9 Square: 3oddElement: 5 16 Square: 4evenElement: 1 0 Square: 0evenElement: 2 1 Square: 1oddElement: 3 4 Square: 2evenElement: 4 9 Square: 3oddElement: 5 16 Square: 4even
```
Comment (tul1) [inline arrays_and_loops/test.txt:1]: what is the propose of this file?

File: arrays_and_loops/arrays_and_loop.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+		arr[i] = i*i
+		var odd string
+		if arr[i]%2==1 {
+			odd="odd"
+		} else {
+			odd="even"
+		} 
+		fmt.Print("\nElement ", i+1, ": ", arr[i], ",  Square: ", math.Sqrt(float64(arr[i])), ", ",  odd, "\n")
```
Comment (tul1) [inline arrays_and_loops/arrays_and_loop.go:18]: Golang is types language, this means that variables are types. for example, `arr` is an array of `int` or `odd` is a `string`. To be printed each type have a formatting verb (doc [link](https://pkg.go.dev/fmt)). can you use the correct verb for the element that you want to print using `Printf` like you did in the other PR?

File: hello_world/README.md [hunk 5]
```diff
@@ -11,7 +11,17 @@ For this first assignment, Edouard needs to write a simple Go program (`hello_wo
 3. Update this README with instructions on how to run the `hello_world.go` code. The instructions should explain how to compile and run the program using Go.
 
 ## Answers
-Via le chemin relatif j’execute ce qui suit : 
-go run hello_world.go (compilation + execution)
-go build helo_world.go (compilation)
-pour exec : ./test (execution)
+compilation:
```
Comment (tul1) [inline hello_world/README.md:14]: This patch belong to another branch and another PR. Let's keep the commit history clean.

**Review by tul1 — CHANGES_REQUESTED** [review]
not there yet, but you are getting closer.

File: arrays_and_loops/arrays_and_loop.go (hunk 2 above)
Comment (busin11) [reply arrays_and_loops/arrays_and_loop.go:10]: for sure, you're right why define a loop to iterate over an array if i have 'range' func used for

File: arrays_and_loops/test.txt (hunk 3 above)
Comment (busin11) [reply arrays_and_loops/test.txt:1]: a test for output result in %#v verb 
not really useful, juste for my comprehension

File: arrays_and_loops/README.md [hunk 6]
```diff
@@ -21,4 +21,27 @@ Element 2: 2, Square: 4, Even
 Element 3: 3, Square: 9, Odd
 Element 4: 4, Square: 16, Even
 Element 5: 5, Square: 25, Odd
-```
\ No newline at end of file
+```
+
+Array Squaring and Even/Odd Identifier
```
[the remark this answers is not in the document]
Comment (busin11) [reply arrays_and_loops/README.md:26]: done

File: arrays_and_loops/README.md (hunk 1 above)
Comment (busin11) [reply arrays_and_loops/README.md:30]: done ; not necessary cause its not a valid file for this assignment

File: arrays_and_loops/arrays_and_loop.go (hunk 4 above)
Comment (busin11) [reply arrays_and_loops/arrays_and_loop.go:18]: done

**Comment by busin11** [comment]
I might need your help cause i've added a pull request with strings_and_functions to add solution and edited readme but when i'am moving to commit sections of pull request it displays me previous commit too.

i'll look deeper into the functionning of git.

File: arrays_and_loops/README.md [hunk 7]
```diff
@@ -21,4 +21,16 @@ Element 2: 2, Square: 4, Even
...
+
+Exercie 2
+   
+create basic array and loop over it to print simple values
+   
+How it works
+1- Initialize an array of 5integer and odd var of type string
+2- Iterate over array with value i
+3- at each iteration, associate index of array with i*i
+4- at each iteration check if modulo 2 of array index is 0 or not, if 0 then associate odd string value to odd var else associate even value to odd var
```
Comment (tul1) [inline arrays_and_loops/README.md:34]:
```suggestion
4- at each iteration it check if the element of the array is even or odd. If `arr[i]%2==1` is true means that the number is odd. Complementary If `arr[i]%2==1` is false the number is pair (or even).
```

File: arrays_and_loops/arrays_and_loop.go [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+	var arr [5]int
+	for i := range arr {
+		arr[i] = i*i
+		var odd string
+		if arr[i]%2==1 {
+			odd="odd"
+		} else {
+			odd="even"
+		} 
+		fmt.Printf("Element %d: %d , Square %#v , %s", i+1, arr[i], math.Sqrt(float64(arr[i])), odd+"\n")
```
Comment (tul1) [inline arrays_and_loops/arrays_and_loop.go:18]:
```suggestion
		fmt.Printf("Element %d: %d , Square %f , %s\n", i+1, arr[i], math.Sqrt(float64(arr[i])), odd) 
```

**Review by tul1 — CHANGES_REQUESTED** [review]
revert the changes in hello_world/README.md and address the other comment and we'll be done :).

File: arrays_and_loops/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (busin11) [reply arrays_and_loops/README.md]: i will check why env is not set correctly

File: hello_world/README.md [hunk 9]
```diff
@@ -11,7 +11,17 @@ For this first assignment, Edouard needs to write a simple Go program (`hello_wo
...
+'''
```
Comment (tul1) [inline hello_world/README.md:15]:
```suggestion
```bash
```

File: hello_world/README.md [hunk 10]
```diff
@@ -11,7 +11,17 @@ For this first assignment, Edouard needs to write a simple Go program (`hello_wo
...
+go build hello_world.go
+'''
```
Comment (tul1) [inline hello_world/README.md:17]:
```suggestion
```
```

File: hello_world/README.md [hunk 11]
```diff
@@ -11,7 +11,17 @@ For this first assignment, Edouard needs to write a simple Go program (`hello_wo
...
+
+execution:
+'''
```
Comment (tul1) [inline hello_world/README.md:20]:
```suggestion
```
```

File: hello_world/README.md [hunk 12]
```diff
@@ -11,7 +11,17 @@ For this first assignment, Edouard needs to write a simple Go program (`hello_wo
...
+go run hello_world.go
+'''
```
Comment (tul1) [inline hello_world/README.md:22]:
```suggestion
```
```

File: hello_world/README.md [hunk 13]
```diff
@@ -11,7 +11,17 @@ For this first assignment, Edouard needs to write a simple Go program (`hello_wo
...
+
+execution:
+'''
```
Comment (tul1) [inline hello_world/README.md:25]:
```suggestion
```bash
```

File: hello_world/README.md [hunk 14]
```diff
@@ -11,7 +11,17 @@ For this first assignment, Edouard needs to write a simple Go program (`hello_wo
...
+./hello_world.go
+'''
```
Comment (tul1) [inline hello_world/README.md:27]:
```suggestion
```
```

**Comment by busin11** [comment]
You've askied me to use the next command in order to not commit previous changes in PR in another PR : "git pull origin main_or_branch --rebase", but it doesn't works for me, maybe i didnt do the right thing ; also i've to set up correctly my env to be sure to work efficently.

I purpose you to send you emails with my assignment solutions & readme for tomorrow ?

File: arrays_and_loops/arrays_and_loop.go [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"fmt"
+	"math"
+)
+
+func main() {
+	var arr [5]int
+	for i := range arr {
+		arr[i] = i*i
+		var odd string
```
Comment (tul1) [inline arrays_and_loops/arrays_and_loop.go:12]:
```suggestion
		var pairity string
```

**Comment by busin11** [comment]
Let me know when i can merge and create another PR for next assignment.

**Pull request merged**